src/main/java/uk/gov/hmcts/reform/cwrdapi/controllers/CaseWorkerRefUsersController.java

```diff
@@ -182,4 +189,61 @@
 
     }
 
+    @ApiOperation(value = "Delete Case Worker Profiles by User ID or Email Pattern",
+            authorizations = {
+                    @Authorization(value = "ServiceAuthorization"),
+                    @Authorization(value = "Authorization")
+            })
+    @ApiResponses({
+            @ApiResponse(
+                    code = 204,
+                    message = "Case Worker Profiles deleted successfully",
+                    response = CaseWorkerProfilesDeletionResponse.class
+            ),
+            @ApiResponse(
+                    code = 400,
+                    message = "An invalid request has been provided"
+            ),
+            @ApiResponse(
+                    code = 401,
+                    message = "Unauthorized Error : The requested resource is restricted and requires authentication"
+            ),
+            @ApiResponse(
+                    code = 403,
+                    message = "Forbidden Error: Access denied"
+            ),
+            @ApiResponse(
+                    code = 500,
+                    message = "Internal Server Error"
+            )
+    })
+
+    @DeleteMapping(
+            produces = APPLICATION_JSON_VALUE
+    )
+    @ResponseStatus(value = HttpStatus.NO_CONTENT)
+    @ResponseBody
+    public ResponseEntity<CaseWorkerProfilesDeletionResponse> deleteCaseWorkerProfileByIdOrEmailPattern(
+            @RequestParam(value = "userId", required = false) String userId,
+            @RequestParam(value = "emailPattern", required = false) String emailPattern) {
+
+        /**
+         * This API will need to be revisited if it is to be used for business functionality.
+         */
+
+        CaseWorkerProfilesDeletionResponse resource;
+
+        if (isNotBlank(userId)) {
+            resource = caseWorkerService.deleteByUserId(userId);
+
+        } else if (isNotBlank(emailPattern)) {
+            resource = caseWorkerService.deleteByEmailPattern(emailPattern);
+
+        } else {
+            throw new InvalidRequestException("No User ID or Email Pattern provided to delete the User(s)");
```

Comment:
I think use existing exception Forbidden with specific message. And avoid hard coded error message.

Reply:
This is in the event of a `400` `Bad Request` so I think Forbidden would not be correct here @abhijitdiwan ?

Reply:
ok ok...replace message with constant.

Reply:
Addressed here: https://github.com/hmcts/rd-caseworker-ref-api/pull/287/commits/4d75068b27e5a480f5dab4627536f5263326fa99